ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -100,6 +100,13 @@ StatusDialog {
 
         function onSelectedNetworkChainIdChanged() {
             networkFilter.selection = [root.swapInputParamsForm.selectedNetworkChainId]
+
+            // Network change needs to be handled manually due to change in available tokens and
+            // relation between pay/from input panels, to ensure consistent results.
+            payPanel.selectedNetworkChainId = root.swapInputParamsForm.selectedNetworkChainId
+            receivePanel.selectedNetworkChainId = root.swapInputParamsForm.selectedNetworkChainId
+            payPanel.reevaluateSelectedId()
```

Comment:
and then the binding above is triggered to reevaluateSelectedId. IMO it's an internal matter for the panels to reevaluate when the input changes.